source/particle/world.cc

```diff
@@ -501,6 +501,53 @@ namespace aspect
     }
 
     template <int dim>
+    bool
+    World<dim>::particle_is_in_cell(const Particle<dim> &particle,
+                                    const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+    {
+      try
+        {
+          const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(cell, particle.get_location());
+          if (GeometryInfo<dim>::is_inside_unit_cell(p_unit))
+            return true;
+        }
+      catch (...)
```

Comment:
Can you try to only catch the one specific exception type that indicates that the point was not found? All other exceptions are then allowed to propagate.